xbmc/filesystem/AddonsDirectory.cpp

```diff
@@ -824,6 +825,12 @@ void CAddonsDirectory::GenerateAddonListing(const CURL& path,
                                                                     addon->Version());
     bool disabled = CServiceBroker::GetAddonMgr().IsAddonDisabled(addon->ID());
 
+    bool isUpdate =
+        std::any_of(updates.begin(), updates.end(), [&](const std::shared_ptr<IAddon>& i) {
+          return i->ID() == addon->ID() && i->Origin() == addon->Origin() &&
```

Comment:
If this a valid check? If the list of updates returned has an update from another repo this will fail. Similar case for outdated addons.

Reply:
Ignore my comment. This is correct, it must always match on those criteria.